mobile/src/navigation/app/components/drawer-content/drawer-content.tsx

```diff
@@ -20,16 +21,28 @@ import { coursesActions } from '~/store/actions';
 
 import { styles } from './styles';
 
-const DrawerContent: FC<DrawerContentComponentProps> = ({ state }) => {
+type Props = {
+  props: DrawerContentComponentProps;
+  navigationItems: DrawerNavigationList[];
+};
+
+const DrawerContent: FC<Props> = ({ props, navigationItems }) => {
   const dispatch = useAppDispatch();
+  const { state } = props;
   const focusedRouteName = state.routes[state.index].name as AppScreenName;
   const allowedRoutes = state.routes.map((item) => item.name);
-  const visibleNavigationItems = NAVIGATION_ITEMS.filter(
-    (item) => item.isVisible,
-  );
 
-  const { isMentorBecomingVisible, dataBecomeMentorStatus } = useAppSelector(
-    (rootState) => rootState.courses,
+  const { user, isMentorBecomingVisible, dataBecomeMentorStatus } =
+    useAppSelector(({ auth, courses }) => ({
+      user: auth.user,
+      isMentorBecomingVisible: courses.isMentorBecomingVisible,
+      dataBecomeMentorStatus: courses.dataBecomeMentorStatus,
+    }));
+
+  const visibleNavigationItems = navigationItems.filter(
+    (item) =>
+      item.isVisible &&
+      Boolean(getScreensByAuth(item.subroutes, Boolean(user)).length),
```

Comment:
Does this check need for us, we have 
```
subroutes={subroutes.filter((item) =>
    allowedRoutes.includes(item.name),
)}
```
allowedRoutes calculated by parent component App(AppNavigator)

Reply:
Or I am wrong?

Reply:
Without this check, we have this empty section in drawer
![Screenshot_1](https://user-images.githubusercontent.com/89215185/189292292-82b9d9f5-5839-4fd7-8467-f8ac37169a69.png)

Reply:
I don’t like this logic at all, we discussed it and came to the following:
1. remove groups from NAVIGATION_ITEMS for drawer. We don't need isVisible more too, but you should add not optional property drawerGroup for DrawerNavigationItem, we will group routes for drawer by it, so add it only for screen that should be shown in drawer. Now it should look like this:
```typescript
// in navigation/app/common/types/drawer-group.type.ts
types DrawerGroup = 'Menu' | 'Account';
```
```typescript
const NAVIGATION_ITEMS = [
      {
        name: AppScreenName.COURSES,
        icon: 'book',
        component: Courses,
        permissions: [],
        drawerGroup:  'Menu',
      },
      {
        name: AppScreenName.UAM_GROUPS_EDIT,
        icon: 'uam',
        component: UAMConfigureGroup,
        permissions: [PermissionKey.MANAGE_UAM],
      },
      ...
];
``` 
2. filter allowed screens with drawerGroup and pass them to DrawerContent, also for DrawerContent item type drawerGroup is required, create a new type for it that will reuse DrawerNavigationItem type and make drawerGroup required
```typescript
const drawerItems = allowedScreens.filter(({ drawerGroup }) => Boolean(drawerGroup));
//
(props): ReactElement => <DrawerContent {...props} items={drawerItems} />
```
3. in DrawerContent remove all the logic with filtering/auth/visibility and so on. The only thing that we will need is grouping, create a common helper for this like groupByKey and then use it in DrawerContent like below:
```typescript
const itemGroups = groupByKey(items, 'drawerGroup'); 
/* should look like this
  {
    Menu: [....your navigation items]
  }
*/
```
4. after this iterate `Object.entries(itemGroups)` to render needed lists and items
5. Also remove isFocused from DrawerNavigationItem, we need to pass it as separate prop